src/Microsoft.DotNet.ProjectModel/TargetLibraryWithAssets.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using Microsoft.DotNet.ProjectModel.Graph;
+using NuGet.Frameworks;
+
+namespace Microsoft.DotNet.ProjectModel
+{
+    public abstract class TargetLibraryWithAssets : LibraryDescription
```

Comment:
I like the current name. It means that this class represents target libraries for which the asset fields exist and make sense. This is the case for msbuild and packages. Library target objects representing project.json projects do not have those asset fields populated.

What about LibraryDescriptionWithAssets?